BlocSpot/app/src/main/java/com/izor066/android/blocspot/ui/adapter/TabAdapter.java

```diff
@@ -32,15 +33,45 @@ public TabAdapter(FragmentManager fm, CharSequence Titles[], int NumbOfTabsumb)
     public Fragment getItem(int position) {
 
         if (position == 0) {
-            PointsOfInterestFragmentList pointsOfInterestFragmentList = new PointsOfInterestFragmentList();
+            pointsOfInterestFragmentList = new PointsOfInterestFragmentList();
             return pointsOfInterestFragmentList;
         } else {
-            PointsOfInterestMapFragment pointsOfInterestMapFragment = new PointsOfInterestMapFragment();
+            pointsOfInterestMapFragment = new PointsOfInterestMapFragment();
             return pointsOfInterestMapFragment;
         }
 
     }
 
+    public void updateCategory(String category) {
+        try {
+            pointsOfInterestFragmentList.updateCategory(category);
+            pointsOfInterestMapFragment.updateCategory(category);
+        } catch (NullPointerException e) {
```

Comment:
`NullPointerException`s should never be caught as they hinder a problem in the code. Why could the fragments be null here? Do you call `updateCategory()` before they're created? 

As a last resort if you want to guard against the null you can do something like this at the beginning of the method (if both of the fragments are null at some point): 

```java
if (pointsOfInterestFragmentList == null || pointsOfInterestMapFragment == null) {
    return;
}
```

The above will stop the method executing if any of the fragment is null. Is there any case in which one of them is null and the other not? In this case the guard could be more granular, as in: 

```java
if (pointsOfInterestFragmentList!=null) {
  pointsOfInterestFragmentList.updateCategory(category);  
}

if (pointsOfInterestMapFragment!=null) {
   pointsOfInterestMapFragment.updateCategory(category); 
}
```

The same goes for the rest of the `try catch` for `NullPointerException`. In any case, catching this exception is very unadvisable.